### DISCUSSION TITLE
Filament Fragment Shader: about adjacent pixel of texture

### DISCUSSION THREAD

**Discussion by JackKaiXing** [google/filament#4431] [Q&A]
Hello there,

How could I accurately and stablely sample adjacent pixel of given UV of a 2D texture.

--------------------------------------------------------------------------------------
Here is what I did.

I create a texture sampler2D with `width=height=2014, level=0`, a textureSampler with ` magfilter=nearest, minfilter=nearest`. 
```
float offset = 1.0f / width;
vec2 UV = getUV0(); // the center pixel
vec2 leftUp = UV + vec2(-offset,+ offset);
vec2 leftMiddle = UV + vec2(-offset,0.0);
vec2 leftBottom = UV + vec2(-offset,-offset);
vec2 RightUp = UV + vec2(offset,offset);
............
```
--------------------------------------------------------------------------------------

Is this the right way to sample current UVs' adjacent `3X3 window`'s pixel in filament. I expect to the sampling results could be as stable as possible. However in my test, it jitters.

**Comment by pixelflinger** [comment]
The best way to do this is to use `textureLodOffset`
